### PR TITLE
Add FluentD chart

### PR DESCRIPTION
**Pull request by rendhalver**
This is not a DaemonSet for Kubernetes logging
It is intended to be a replacement for logstash in an elastic-stack cluster

*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

### CODE REVIEW

File: incubator/fluentd/templates/deployment.yaml [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+          - name: OUTPUT_PORT
+            value: {{ .Values.output.port | quote }}
+          - name: OUTPUT_BUFFER_CHUNK_LIMIT
+            value: {{ .Values.output.buffer_chunk_limit | quote }}
+          - name: OUTPUT_BUFFER_QUEUE_LIMIT
+            value: {{ .Values.output.buffer_queue_limit | quote }}
+        resources:
+{{ toYaml .Values.resources | indent 12 }}
+        ports:
+{{- range $port := .Values.service.ports }}
```
Comment (unguiculus) [inline incubator/fluentd/templates/deployment.yaml:44]: I'm not familiar with FluentD configuration. How is this supposed to work? Wouldn't you have to configure these ports in FluentD somehow? This doesn't seem to happen.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: incubator/fluentd/templates/deployment.yaml (hunk 1 above)
Comment (rendhalver) [reply incubator/fluentd/templates/deployment.yaml:44]: The monitor-agent port is configured in the configMap in values.
The entire config is in values and source configs in FluentD are pretty varied so configuring the ports in the FluentD config would be up to the user.

Comment (rendhalver) [reply incubator/fluentd/templates/deployment.yaml:44]: The ports in values are setup in the default config.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**